### PR TITLE
try to implement FilterAdapter

### PR DESCRIPTION
**Pull request by basakest**
try to implement FilterAdapter

### CODE REVIEW

**Comment by hsluoyz** [comment]
Fix: https://github.com/php-casbin/database-adapter/issues/4

@pigochu @techoner please review.

File: index.php [hunk 1]
```diff
@@ -0,0 +1,36 @@
+<?php
```
Comment (hsluoyz) [inline index.php]: Why add this?

File: src/FilterAdapter.php [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+            $line = implode(', ', $row);
+            $this->loadPolicyLine($line, $model);
+        }
+        
+        $this->filtered = true;
+    }
+
+    /**
+     * load filtered policy
+     * 通过拼接sql语句来实现过滤功能，但直接将字段的值进行了拼接，存在安全方面的问题，无法防止SQL注入等情况，应该用query方法的第二个参数来改写
```
Comment (hsluoyz) [inline src/FilterAdapter.php]: I think this repo uses English?

**Review by hsluoyz — CHANGES_REQUESTED** [review]

**Comment by leeqvip** [comment]
@basakest It is recommended to implement `FilterAdapter` directly on `CasbinAdapter\Database\Adapter`, refer to [DBAL Adapter](https://github.com/php-casbin/dbal-adapter), then please add unit test.

**Pull request merged**